src/pages/blog/experimental-server-side-rendering.md

```diff
@@ -0,0 +1,171 @@
+- Dynamic sites can’t pre-render to static HTML without forcing you to rebuild and redeploy some portion of your site whenever your data changes.
+- Interactive sites can’t pre-render to static HTML without moving all user-individualized data to the client. User names, avatars, and permissions all get more complicated when you’re dealing with static, pre-built HTML.
+
+E-commerce is especially hurt by all three of the above problems. In an industry where every millisecond of load performance counts, how can you get the awesome performance of static HTML without the limitations?
+
+From Day 1, we knew that Astro’s approach to island architecture was uniquely positioned to solve this problem. Islands allow pages to load quickly and become interactive lazily; the user sees and is able to interact with the most important parts of the page, faster.
+
+# Enter: Server-Side Rendering (SSR)
+
+Server-side rendering is a well-understood solution to scaling large websites, going back to the earliest days of the internet. SSR is no silver bullet, but, when done right, SSR is an invaluable tool for certain use-cases.
+
+Take user authentication, which is something not easily achieved on a static site. You might do a couple of different ways:
+
+1. Check for a cookie or JWT in localStorage, then redirect to a login page if the user doesn’t have one. If using a cookie this prevents using `HttpOnly`, opening you to malicious JavaScript attacks. Also not shown here is, what do you do if the token is not valid? You still need to check and respond in that case.
+
+```html
+<script>
+  if(!parseCookie(document.cookie).auth) {
+    window.location = '/login';
+  }
+</script>
+```
+
+1. Call an API that will check if a user is logged in, like `/api/auth`. Going this route is going to mean that you need to show loading spinners in each of your islands while you wait on the response.
+
+```jsx
+function App() {
+  const auth = useAuth();
+  if(auth.loading) return <AnnoyingLoadingSpinner />
+  if(!auth.loggedIn) return <LoginPage />;
+  return <UserProfile user={auth.user} />
+}
+```
+
+No matter your approach, going client-side for authentication is always going to mean:
+
+- Things won’t run if JavaScript is disabled.
+- Your server can only respond with `200 OK` response codes.
+- It’s just plain slow. The user has to wait for the full HTML response (including JS, CSS, etc) even if they then immediately redirect. This creates multiple page requests before the user lands on the correct page.
+
+## Astro SSR: Designed to be Simple
+
+SSR might sound like a solved problem. Next.js, Nuxt, Gatsby, SvelteKit, and the whole gang of modern JavaScript meta-frameworks all have some idea of SSR already in place. SSR as a term is really just a rebranding of how Rails, PHP and other server frameworks already work today. If that’s all true, then what’s the big deal?
```

Comment:
```suggestion
Next.js, Nuxt, Gatsby, SvelteKit, and the whole gang of modern JavaScript meta-frameworks have some idea of SSR already. So what makes Astro SSR so special?
```

Copied from the Notion feedback! Curious if we can shorten this lead-in to be more Astro-centric.